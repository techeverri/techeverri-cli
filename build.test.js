import { describe, test, expect } from "bun:test";
import fs from "fs";

describe("build", () => {
  test("should generate card with colors", () => {
    expect.assertions(1);

    const card = fs.readFileSync("./bin/card", { encoding: "utf8", flag: "r" });

    expect(card).toMatchSnapshot();
  });
});
