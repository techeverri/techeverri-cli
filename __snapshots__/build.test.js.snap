// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`build should generate card with colors 1`] = `
"\x1B[32m\x1B[39m
\x1B[32m   ┌────────────────────────────────────────────────────────────────┐\x1B[39m
\x1B[32m   │                                                                │\x1B[39m
\x1B[32m   │   \x1B[37mTomas Echeverri\x1B[39m\x1B[32m / \x1B[37mtecheverri\x1B[39m\x1B[32m                                 │\x1B[39m
\x1B[32m   │   \x1B[90m\x1B[3mMade in \x1B[33mColo\x1B[39m\x1B[32m\x1B[90m\x1B[34mmb\x1B[39m\x1B[32m\x1B[90m\x1B[31mia\x1B[39m\x1B[32m\x1B[90m. Living in Sweden.\x1B[23m\x1B[39m\x1B[32m                          │\x1B[39m
\x1B[32m   │                                                                │\x1B[39m
\x1B[32m   │        \x1B[37m\x1B[1mWork:\x1B[22m\x1B[39m\x1B[32m \x1B[37mSenior Software Developer at \x1B[32m\x1B[3mShopify\x1B[23m\x1B[39m\x1B[32m\x1B[37m\x1B[39m\x1B[32m              │\x1B[39m
\x1B[32m   │           \x1B[37m\x1B[1mX:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://x.com/\x1B[36mTomasEcheverri\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m                      │\x1B[39m
\x1B[32m   │    \x1B[37m\x1B[1mMastodon:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://mastodon.social/\x1B[35m@techeverri\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m               │\x1B[39m
\x1B[32m   │     \x1B[37m\x1B[1mBluesky:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://bsky.app/profile/\x1B[34mtecheverri.bsky.social\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m   │\x1B[39m
\x1B[32m   │      \x1B[37m\x1B[1mGitHub:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://github.com/\x1B[32mtecheverri\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m                     │\x1B[39m
\x1B[32m   │         \x1B[37m\x1B[1mnpm:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://www.npmjs.com/\x1B[31m~techeverri\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m                 │\x1B[39m
\x1B[32m   │    \x1B[37m\x1B[1mLinkedIn:\x1B[22m\x1B[39m\x1B[32m \x1B[90mhttps://www.linkedin.com/in/\x1B[34mtomechval\x1B[39m\x1B[32m\x1B[90m\x1B[39m\x1B[32m             │\x1B[39m
\x1B[32m   │     \x1B[37m\x1B[1mWebsite:\x1B[22m\x1B[39m\x1B[32m \x1B[35mhttps://techeverri.dev\x1B[39m\x1B[32m                            │\x1B[39m
\x1B[32m   │                                                                │\x1B[39m
\x1B[32m   │        \x1B[37m\x1B[1mCard:\x1B[22m\x1B[39m\x1B[32m npx techeverri                                    │\x1B[39m
\x1B[32m   │                                                                │\x1B[39m
\x1B[32m   └────────────────────────────────────────────────────────────────┘\x1B[39m
\x1B[32m\x1B[39m"
`;
